esrally/client.py

```diff
@@ -25,16 +25,15 @@ def __init__(self, hosts, client_options):
         # we're using an SSL context now and it is not allowed to have use_ssl present in client options anymore
         if self.client_options.pop("use_ssl", False):
             import ssl
-            from elasticsearch.connection import create_ssl_context
             self.logger.info("SSL support: on")
             self.client_options["scheme"] = "https"
 
-            self.ssl_context = create_ssl_context(cafile=self.client_options.pop("ca_certs", certifi.where()))
+            self.ssl_context = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH, cafile=self.client_options.pop("ca_certs", certifi.where()))
```

Comment:
This is required to enable client authentication, if requested by the server.

Using `ssl.create_default_context`, because the one in [elasticsearch-py](https://github.com/elastic/elasticsearch-py/blob/master/elasticsearch/connection/http_urllib3.py#L27-L36) is a simple wrapper for it, but only passes kwargs and won't pass `ssl.Purpose.CLIENT_AUTH`.